Add MyReviews page

j1yl opened the pull request:
TODO

j1yl commented:
> <img alt="Screenshot 2025-05-23 at 6 25 02 PM" width="1429" src="https://private-user-images.githubusercontent.com/114836433/447206438-7dd1a144-cfa7-4bfa-bbc3-5a45dd8aaf1f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwNTMxNjEsIm5iZiI6MTc0ODA1Mjg2MSwicGF0aCI6Ii8xMTQ4MzY0MzMvNDQ3MjA2NDM4LTdkZDFhMTQ0LWNmYTctNGJmYS1iYmMzLTVhNDVkZDhhYWYxZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUyNFQwMjE0MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYjcwYzcwM2ZlODI0YWFhYzAyMDY1ZDcyMmRhYzZkNzY1NDYwZDM3YjA4MmM5Zjc4ZTdhY2E3YTZlYWEzNjFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.NnpvPG59WKCM9Q2TIre7CJPto3UMLy1vjTC8lo8a4Ls">
> 
> I am trying to add reviews on swagger and then seeing the changes on the review page, but for some reason a lot of the information isn't showing?

Issue was the mismatch between the accessor name and actual name returned by the API response, I updated the name in the files using them, issue was fixed, can be confirmed here https://dining-proj-j1yl.dokku-04.cs.ucsb.edu/myreviews

To test, just POST a test review and navigate to `/myreviews`

![image](https://github.com/user-attachments/assets/837d1655-0ee6-421c-8967-0888c21a9eca)

ZFahim10 reviewed (approved):
I was able to view the post I added through swagger> the also editing worked. At the end, I was also able to delete the post. Good Job

j1yl commented:
## Changelog
1. Added support for primitive values in `ReviewTable.js`, this fixed the blank column of item id
3. Changed route parameters in `ReviewForMenuItemPage.js` and `ReviewForMenuItemPage.test.js` from `itemid` to `id` to match the backend.

## To test
Sign in to https://dining-proj-j1yl.dokku-04.cs.ucsb.edu/

Navigate to https://dining-proj-j1yl.dokku-04.cs.ucsb.edu/reviews/417

![image](https://github.com/user-attachments/assets/735c16f7-f59f-4f1c-8402-a933c3273391)
